bindings/python/src/manifest.rs

```diff
@@ -0,0 +1,420 @@
+        self.inner.value_counts()
+    }
+
+    #[getter]
+    fn null_value_counts(&self) -> &HashMap<i32, u64> {
+        self.inner.null_value_counts()
+    }
+
+    #[getter]
+    fn nan_value_counts(&self) -> &HashMap<i32, u64> {
+        self.inner.nan_value_counts()
+    }
+
+    #[getter]
+    fn upper_bounds(&self) -> HashMap<i32, Vec<u8>> {
+        self.inner.upper_bounds().into_iter().map(|(k, v)| (k.clone(), v.to_bytes().unwrap().to_vec())).collect()
+    }
+
+    #[getter]
+    fn lower_bounds(&self) -> HashMap<i32, Vec<u8>> {
+        self.inner.lower_bounds().into_iter().map(|(k, v)| (k.clone(), v.to_bytes().unwrap().to_vec())).collect()
+    }
+
+    #[getter]
+    fn key_metadata(&self) ->  Option<&[u8]> {
+        self.inner.key_metadata()
+    }
+
+    #[getter]
+    fn split_offsets(&self) -> &[i64] {
+        self.inner.split_offsets()
+    }
+
+    #[getter]
+    fn equality_ids(&self) -> &[i32] {
+        self.inner.equality_ids()
+    }
+
+    #[getter]
+    fn sort_order_id(&self) -> Option<i32> {
+        self.inner.sort_order_id()
+    }
+
+}
+
+#[pyclass]
+pub struct PyManifest {
+    inner: Manifest,
+}
+
+
+#[pymethods]
+impl PyManifest {
+    fn entries(&self) -> Vec<PyManifestEntry> {
+        // TODO: Most of the time, we're only interested in 'alive' entries,
+        // that are the ones that are either ADDED or EXISTING
+        // so we can add a boolean to skip the DELETED entries right away before
+        // moving it into the Python world
+        self.inner.entries().iter().map(|entry| PyManifestEntry { inner: entry.clone() }).collect()
+    }
+}
+
+
+#[pyclass]
+pub struct PyFieldSummary {
+    inner: FieldSummary,
+}
+
+
+#[pymethods]
+impl crate::manifest::PyFieldSummary {
+
+    #[getter]
+    fn contains_null(&self) -> bool {
+        self.inner.contains_null
+    }
+
+    #[getter]
+    fn contains_nan(&self) -> Option<bool> {
+        self.inner.contains_nan
+    }
+
+    #[getter]
+    fn lower_bound(&self) -> Option<PyPrimitiveLiteral> {
+        self.inner.lower_bound.clone().map(|v| PyPrimitiveLiteral{ inner: v.literal().clone() })
+    }
+
+    #[getter]
+    fn upper_bound(&self) -> Option<PyPrimitiveLiteral> {
+        self.inner.upper_bound.clone().map(|v| PyPrimitiveLiteral{ inner: v.literal().clone() })
+    }
+
+
+
+}
+
+#[pyclass]
+pub struct PyManifestFile {
+    inner: ManifestFile,
+}
+
+
+#[pymethods]
+impl crate::manifest::PyManifestFile {
+    #[getter]
+    fn manifest_path(&self) -> &str {
+        self.inner.manifest_path.as_str()
+    }
+    #[getter]
+    fn manifest_length(&self) -> i64 {
+        self.inner.manifest_length
+    }
+    #[getter]
+    fn partition_spec_id(&self) -> i32 {
+        self.inner.partition_spec_id
+    }
+
+    #[getter]
+    fn content(&self) -> i32 {
+        // TODO: Maps the enum back to the int
+        // I have a feeling this can be done more intelligently
+        match self.inner.content {
```

Comment:
I think you can just do this?

```rust
#[getter]
fn content(&self) -> i32 {
    self.inner.content as i32
}
```

(See https://doc.rust-lang.org/reference/items/enumerations.html#casting)